### ISSUE TITLE
Cannot read property 'on' of null

### ISSUE THREAD

**Issue by alinnert** [felixhao28/node-systray#1]
First of all, I'm super happy you continue this project. I'm still very interested in this.

I've tried out your version of the library but already the demo code causes this error:

~~~
C:\projects\systray-test\node_modules\systray2\lib\index.js:184
        this._rl.on("line", function (line) {
                 ^

TypeError: Cannot read property 'on' of null
    at SysTray.onClick (C:\projects\systray-test\node_modules\systray2\lib\index.js:184:18)
    at Object.<anonymous> (C:\projects\systray-test\index.js:38:9)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
~~~

**Comment by felixhao28** [comment]
Sorry, I forgot to mention I added an "init" method to call before actually spawning the tray process. The point of using an "init" function is to provide an async way of starting the tray process.

```js
async function startTray() {
    const systray = new SysTray({...});
    systray.onClick(...);
    await systray.init();
}
```

**Comment by felixhao28** [comment]
On second thought, I am going to change the behavior to not requiring to call "Systray.init()" but instead provide a "Systray.ready()" to wait for the tray to be ready,

**Comment by felixhao28** [comment]
This should be fixed in v2.0.4 . @alinnert will you try again?

**Comment by alinnert** [comment]
Awesome, it works now. Thanks a lot.
